src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,145 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/Memory.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(std::move(generators))
+    , mIndex(0)
+  {
+    for (std::size_t i = 0; i < mGenerators.size(); ++i)
+    {
+      if (!mGenerators[i])
+      {
+        std::stringstream msg;
+        msg << "Generator " << i << " is nullptr.";
+        throw std::invalid_argument(msg.str());
+      }
+    }
+
+#ifndef NDEBUG
+    for (const auto& generator : mGenerators)
+      assert(generator->getStateSpace() == mStateSpace);
+#endif
+  }
+
+  // Documentation inherited
+  statespace::StateSpacePtr getStateSpace() const override
+  {
+    return mStateSpace;
+  }
+
+  // Documentation inherited
```

Comment:
Nit: `//==============================================================================`, not `// Documentation inherited`.